### DISCUSSION TITLE
Error 500 + Error in/with uninstalled Plugin CustomCSSBundle

### DISCUSSION THREAD

**Discussion by maurice-kpw** [kimai/kimai#4050] [On-Premise: Installation & Updates]
### Describe the issue

I was following the upgrade guide and got stuck after "composer install --optimize-autoloader -n"
We were upgrading from 1.30.10 to the newest version 2.0.22

I uninstalled (deleted) all Plugins from /var/plugins and set the PHP Version to 8.1.19 on my hosting.

Then (and now with every kimai-command) I get the following error:
```Executing script cache:clear [KO]
 [KO]
Script cache:clear returned with error code 1
!!  
!!  In PluginMetadata.php line 59:
!!                                                                                 
!!    Bundle "CustomCSSBundle" defines an invalid Kimai minimum version in extra.  
!!    kimai.require. Please provide an integer as in Constants::VERSION_ID.        
!!                                                                                 
!!  
!!  
Script @auto-scripts was called via post-install-cmd
```

I tried reuploading the old plugin folder, uploading the most recent plugin version and tried without any plugins. All resulting in the same error. 

I don't understand why the Bundle is requiring something after I uninstalled it? 

I found discussion [3086](https://github.com/kimai/kimai/discussions/3086) which has a similar error – the solution was to install database tables for a plugin we don't use / never used.

What am I missing here? Any idea what to do?
Our Installation is not available with an error 500 but I guess thats because of the upgrade in process.

### I already tried

- [X] I've read and searched [the documentation](https://www.kimai.org/documentation/).
- [X] I've searched for similar issues in this repository.
- [X] I've searched for similar issues in [the discussions](https://github.com/kimai/kimai/discussions).

### Kimai version

2.0.22 (?)

### How do you run Kimai?

Shared-Hosting

### Which PHP version are you using?

8.1

### Logfile

No recent error in error log.

**Comment by maurice-kpw** [comment]
Currently I think the problem is that I forgot to clear/reload the cache after uninstalling the plugins and before starting the upgrade process. :(

I just tried the rm -r var/cache/prod/* command this didn't throw the above error but the error still appears when using kimai-commands. 

Is there anything I can do?

**Comment by kevinpapst** [comment]
If you deleted all plugins and `var/plugins/` is empty, then this error can only come from the cache.

You have to delete ALL cache files with `rm -r var/cache/*` and then the error should be gone.

**Reply by maurice-kpw** [reply]
Hello Kevin, 

thanks for getting back to me. 
Yes, the only file in `var/plugins/` is .gitkeep. No other directories.

When trying the command you provided the following error occured:
```
rm: das Entfernen von 'var/cache/pro_' ist nicht möglich: Is a directory
rm: das Entfernen von 'var/cache/prod' ist nicht möglich: Is a directory
```

Did you mean `rm -r var/cache/*`?

– Maurice

**Reply by kevinpapst** [reply]
Yes I did.

**Reply by maurice-kpw** [reply]
Ah thanks for clarifying!

That command worked – but the error still occurs. When I enter `rm -r var/cache/*` again "no such files or directories" is shown. In filezilla I still see directories and files in this folder though. Should I try to delete them manually via filezilla?

**Reply by kevinpapst** [reply]
I don't know. You have to get rid of the cache directories. 
If they are gone and the error still occurs, then you might have opcache enabled, which should never be the case for PHP on CLI.

**Reply by maurice-kpw** [reply]
I see. Manual deletion of the directories didn't work either. But you're right opcache is enabled.
I'm gonna check in with our hosting to get this disabled as I can't switch this off in our plesk administration for our kimai installation.

**Reply by kevinpapst** [reply]
Opcache is good! 
Opcache is only bad for CLI.
You might have to use sudo for deleting the files.

**Reply by maurice-kpw** [reply]
Oh, okay. The problem is I can't use sudo because it's a "changedRoot–Environment".
Is there another way? I'm kinda confused because I deleted the files in 2 ways (manual via filezilla, and via the command you provided) and I don't see them anymore – what is there to delete?

**Reply by kevinpapst** [reply]
You wrote 
> "no such files or directories" is shown. In filezilla I still see directories and files in this folder though.
> I see. Manual deletion of the directories didn't work either. But you're right opcache is enabled.

How shall I know that they are gone?

So cache is empty. What does `php -i|grep opcache.enable` spit out? It should be:
```bash

opcache.enable => On => On
opcache.enable_cli => Off => Off
```

**Reply by maurice-kpw** [reply]
Sorry, my answer wasn't clear enough. You're right.

The deletion worked. I don't see files in the directory var/cache/ anymore. But it didn't solve the problem.

> What does php -i|grep opcache.enable spit out?

Could you clarify what to do with this? When entered in the SSH client (`/opt/plesk/php/8.1/bin/php php -i|grep opcache.enable`) I don't get a result.

**Reply by kevinpapst** [reply]
<img width="398" alt="Bildschirmfoto 2023-05-23 um 17 15 30" src="https://github.com/kimai/kimai/assets/533162/12927806-de1b-4f94-92ae-b48b740373f3">

**Reply by maurice-kpw** [reply]
<img width="934" alt="Bildschirmfoto 2023-05-23 um 17 19 37" src="https://github.com/kimai/kimai/assets/107987851/7d4c5c98-5b7e-448b-b561-72aa249cb67a">
These are the values from the php-info-file. These are the results you were hoping for, right?

**Reply by kevinpapst** [reply]
Cache is gone, opcache is out and you still see the error via SSH?
Show me a screenshot of (do not add or remove something):
```
ls -la var/plugins/
ls -la var/cache/
bin/console kimai:version
ls -la var/cache/
composer install --optimize-autoloader -n
```

**Reply by maurice-kpw** [reply]
Weird. You're absolutely right. Although I deleted the plugins via filezilla it seems they are still there. I don't get it. They show up with the commands you provided, they show up when i go into the web ftp of my hosting but they are invisible with filezilla. 

After moving the plugin-directories with the web-ftp-client to the .disabled-folder everything worked and i could complete the upgrade without issue. 

Thank you so much for your patience and help with that!
